v2/tools/generator/internal/astbuilder/slices.go

```diff
@@ -24,6 +24,24 @@ func MakeSlice(listType dst.Expr, len dst.Expr) *dst.CallExpr {
 	}
 }
 
+// MakeAppendSlice returns the call expression for making a slice with a specified length that can be
+// appended to using append(...)
+//
+// make([]<value>, 0, <len>)
+func MakeAppendSlice(listType dst.Expr, len dst.Expr) *dst.CallExpr {
```

Comment:
Not overly happy with this name.

What about this?

```suggestion
func MakeSliceWithCapacity(listType dst.Expr, capacity dst.Expr) *dst.CallExpr {
```

Reply:
I don't love the name either. With that said, both `MakeSlice` functions are making a slice with a `Capacity`, because the parameter we are calling `len` here is actually always capacity, see: https://go.dev/tour/moretypes/11.

I've changed the `len` argument here to `capacity` on both functions to correct that issue.

That leaves the question of what the name of this function should be... technically we're making a zero length slice with a capacity... I've changed this function name to `MakeSliceZeroLength`. It still has the doc comment that explains "MakeSliceZeroLength returns the call expression for making a slice with a specified length that can be appended to using append(...)"

I don't love this name either but I guess it's at least technically correct. Thoughts?

Reply:
How about `MakeEmptySlice()`?